providers/implementations/encode_decode/decode_der2key.c

```diff
@@ -89,6 +89,7 @@ struct keytype_desc_st {
  */
 struct der2key_ctx_st {
     PROV_CTX *provctx;
+    char *propq;
```

Comment:
Considering that you use a max size, wouldn't it make sense to make this an array?

``` C
    char propq[OSSL_MAX_PROPQUERY_SIZE + 1];
```

(me, I'm thinking "one less malloc, one less fragmentation, one less free")

Reply:
Why impose a max size?  That's just asking for problems in the future.

Ugg, I didn't notice the PR that introduces a constant for max propq size :disappointed:
Which is rarely used.

Reply:
But it is used in multiple places.. Wouldnt removing the malloc be a good thing?

Reply:
OSSL_MAX_PROPQUERY_SIZE is relatively large also..

Reply:
We might want to move towards more preallocation for encoders/decoders...